ui-service/src/main/frontend/hotel-booking-ui/src/components/search-components/HotelSearchSectionComponent.tsx

```diff
@@ -1,257 +1,280 @@
-        const {name, value, type, checked} = e.target;
-        setFormData((prevState) => ({
-            ...prevState,
-            [name]: type === "checkbox" ? checked : value,
-        }));
-    };
+    window.addEventListener("scroll", handleScroll);
 
-    const handleDestinationChange = (
-        index: number,
-        e: React.ChangeEvent<HTMLInputElement>
-    ) => {
-        setFormData(prevState => ({
-            ...prevState,
-            destinationAndDates: prevState.destinationAndDates.map((item, i) =>
-                i === index ? {...item, destination: e.target.value} : item
-            ),
-        }));
+    return () => {
+      window.removeEventListener("scroll", handleScroll);
     };
+  }, []);
 
+  const handleDateChange = (
+    index: number,
+    dates: [Date | null, Date | null]
+  ) => {
+    const [start, end] = dates;
+    setFormData((prevState) => ({
+      ...prevState,
+      destinationAndDates: prevState.destinationAndDates.map((item, i) =>
+        i === index
+          ? {
+              ...item,
+              checkInDate: start || undefined,
+              checkOutDate: end || undefined,
+            }
+          : item
+      ),
+    }));
+  };
 
-    const handleHotelTypeChange = (type: "single" | "multi") => {
-        if (type === "multi") {
-            setFormData((prevState) => ({
-                ...prevState,
-                destinationAndDates: [
-                    {destination: "", startDate: today, endDate: tomorrow},
-                    {
-                        destination: "",
-                        startDate: undefined,
-                        endDate: undefined,
-                    },
-                ],
-                hotelType: type,
-            }));
-        } else {
-            setFormData((prevState) => ({
-                ...prevState,
-                destinationAndDates: [
-                    {destination: "", startDate: today, endDate: tomorrow},
-                ],
-                hotelType: type,
-            }));
-        }
-    };
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value, type, checked } = e.target;
+    setFormData((prevState) => ({
+      ...prevState,
+      [name]: type === "checkbox" ? checked : value,
+    }));
+  };
 
-    const handleChildAgeChange = (childId: number, age: number | null) => {
-        setFormData(prevState => ({
-            ...prevState,
-            childrenAges: prevState.childrenAges.map(child =>
-                child.id === childId ? { ...child, age } : child
-            )
-        }));
-    };
+  const handleDestinationChange = (
+    index: number,
+    e: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setFormData((prevState) => ({
+      ...prevState,
+      destinationAndDates: prevState.destinationAndDates.map((item, i) =>
+        i === index
+          ? {
+              ...item,
+              location: {
```

Comment:
The current implementation of this search field serves as a placeholder for the autocomplete functionality. In the completed version, Priceline uses an autocomplete search component that sends requests to the backend as the user types. The backend responds with an array of location objects from which the user can make a selection.

At this stage, to enable the component to interact with the backend, the location object is populated with minimal information—only the city field based on the user's input. This is a temporary setup, and the autocomplete logic will eventually replace this placeholder.

Reply:
I have couple questions regarding the above: 1. How is frontend supposed to autocomplete the neighborhoodName, which sounds to me a backend data? 2. How is frontend supposed to autocomplete the single lat/long? For example, if a user entered Baltimore, MD, it represents a region which is a set of lat/long pairs to define the region instead of a single lat and long.

Reply:
I think it looks like this. When user typing "Ba", the frontend send an request to backend, which then return a list of location object that is start with "Ba" or somehow related with "Ba". For example, below is part of the json returned by Priceline after I typing "Ba":
```
{
    "resultCode": 0,
    "serverID": "www.priceline.com",
    "resultMessage": "SUCCESS",
    "jsk": null,
    "lang": null,
    "duration": 2,
    "searchItems": [
        {
            "itemName": "Baltimore, MD",
            "id": "3000008983",
            "cityID": "3000008983",
            "type": "CITY",
            "lat": 39.2903,
            "lon": -76.6125,
            "proximity": 0.0,
            "savedTravelStartDate": null,
            "savedTravelEndDate": null,
            "cityName": "Baltimore",
            "stateCode": "MD",
            "provinceName": "Maryland",
            "score": 395746.22,
            "radius": 14.8,
            "rank": 49.659023,
            "rank2": 87.02545928955078,
            "globalScore": 5743.0,
            "globalScoreReducedCityBoost": 34.422333,
            "globalScoreWOHotelCountNormalize": 32.349445,
            "country": "US",
            "address": null,
            "zip": null,
            "countryName": "UNITED STATES",
            "countryCode": "US",
            "poiCategoryTypeId": 0,
            "poiCategoryName": null,
            "poiID": null,
            "seType": null,
            "gmtOffset": -5.0,
            "entered": "ba",
            "highlightedName": null,
            "displayLine1": "Baltimore",
            "displayLine2": "Maryland, United States",
            "fromSavedSearch": false
        },
        {
            "itemName": "Tampa, FL",
            "id": "3000003431",
            "cityID": "3000003431",
            "type": "CITY",
            "lat": 27.9472,
            "lon": -82.4586,
            "proximity": 0.0,
            "savedTravelStartDate": null,
            "savedTravelEndDate": null,
            "cityName": "Tampa",
            "stateCode": "FL",
            "provinceName": "Florida",
            "score": 394583.94,
            "radius": 21.6,
            "rank": 62.94455,
            "rank2": 92.95052337646484,
            "globalScore": 5784.0,
            "globalScoreReducedCityBoost": 42.76517,
            "globalScoreWOHotelCountNormalize": 37.53109,
            "country": "US",
            "address": null,
            "zip": null,
            "countryName": "UNITED STATES",
            "countryCode": "US",
            "poiCategoryTypeId": 0,
            "poiCategoryName": null,
            "poiID": null,
            "seType": null,
            "gmtOffset": -5.0,
            "entered": "ba",
            "highlightedName": null,
            "displayLine1": "Tampa",
            "displayLine2": "Florida, United States",
            "fromSavedSearch": false
        },
        {
            "itemName": "Baton Rouge, LA",
            "id": "3000008192",
            "cityID": "3000008192",
            "type": "CITY",
            "lat": 30.4506,
            "lon": -91.1544,
            "proximity": 0.0,
            "savedTravelStartDate": null,
            "savedTravelEndDate": null,
            "cityName": "Baton Rouge",
            "stateCode": "LA",
            "provinceName": "Louisiana",
            "score": 391477.25,
            "radius": 13.3,
            "rank": 37.324917,
            "rank2": 82.84373474121094,
            "globalScore": 5645.0,
            "globalScoreReducedCityBoost": 34.1276,
            "globalScoreWOHotelCountNormalize": 32.1048,
            "country": "US",
            "address": null,
            "zip": null,
            "countryName": "UNITED STATES",
            "countryCode": "US",
            "poiCategoryTypeId": 0,
            "poiCategoryName": null,
            "poiID": null,
            "seType": null,
            "gmtOffset": -6.0,
            "entered": "ba",
            "highlightedName": null,
            "displayLine1": "Baton Rouge",
            "displayLine2": "Louisiana, United States",
            "fromSavedSearch": false
        },
        {
            "itemName": "Montego Bay, Jamaica",
            "id": "5000025185",
            "cityID": "5000025185",
            "type": "CITY",
            "lat": 18.46936,
            "lon": -77.921387,
            "proximity": 0.0,
            "savedTravelStartDate": null,
            "savedTravelEndDate": null,
            "cityName": "Montego Bay",
            "stateCode": null,
            "provinceName": "Saint James",
            "score": 390215.8,
            "radius": 100.0,
            "rank": 40.857372,
            "rank2": 88.34259033203125,
            "globalScore": 3849.0,
            "globalScoreReducedCityBoost": 30.997833,
            "globalScoreWOHotelCountNormalize": 30.574242,
            "country": "JAMAICA",
            "address": null,
            "zip": null,
            "countryName": "JAMAICA",
            "countryCode": "JM",
            "poiCategoryTypeId": 0,
            "poiCategoryName": null,
            "poiID": null,
            "seType": null,
            "gmtOffset": -5.0,
            "entered": "ba",
            "highlightedName": null,
            "displayLine1": "Montego Bay",
            "displayLine2": "Jamaica",
            "fromSavedSearch": false
        },
...
```
If I keep typing "Baltimore Maryland", the returned list may even contain hotel in the region:
```
{
            "itemName": "Hotel Brexton, Trademark Collection BY Wyndham - Baltimore, MD",
            "id": "9997003",
            "cityID": "3000008983",
            "type": "HOTEL",
            "lat": 39.301173,
            "lon": -76.618613,
            "proximity": 0.0,
            "savedTravelStartDate": null,
            "savedTravelEndDate": null,
            "cityName": "Baltimore",
            "stateCode": "MD",
            "provinceName": "Maryland",
            "score": 400670.9,
            "radius": 14.8,
            "rank": 11.928457,
            "rank2": 97.82095336914062,
            "globalScore": 1.0,
            "globalScoreReducedCityBoost": 1.0,
            "globalScoreWOHotelCountNormalize": 1.0,
            "country": "US",
            "address": "868 Park Avenue",
            "zip": "21201",
            "countryName": "UNITED STATES",
            "countryCode": "US",
            "poiCategoryTypeId": 0,
            "poiCategoryName": null,
            "poiID": null,
            "seType": null,
            "gmtOffset": -5.0,
            "entered": "baltimore maryland",
            "highlightedName": null,
            "displayLine1": "Hotel Brexton, Trademark Collection BY Wyndham",
            "displayLine2": "Baltimore, Maryland, United States",
            "fromSavedSearch": false
        },
```
Basically everything is fetched from the backend, and the frontend only sends the request and let the user to select the location after typing. For the data source, we can either create a mocking API or use a free open geographical API.

Reply:
Doesn't that mean we only keep the address, such as city, state and country for users to select, in other words, how are lat, long, neighborhoodName used in the search form?
<img width="474" alt="image" src="https://github.com/user-attachments/assets/372bf759-1d20-4129-a5f1-7e0d1d653d21">

Reply:
Well, neighborhoodName may be used when user typing something like "Los Angeles Downtown", which is not exactly a city. 
```
{
    "resultCode": 0,
    "serverID": "www.priceline.com",
    "resultMessage": "SUCCESS",
    "jsk": null,
    "lang": null,
    "duration": 5,
    "searchItems": [
        {
            "itemName": "Downtown Los Angeles - Los Angeles, CA",
            "id": "99160106",
            "cityID": "3000001947",
            "type": "POI",
            "lat": 34.051,
            "lon": -118.254333,
            "proximity": 0.0,
            "savedTravelStartDate": null,
            "savedTravelEndDate": null,
            "cityName": "Los Angeles",
            "stateCode": "CA",
            "provinceName": "California",
            "score": 168329.11,
            "radius": 22.3,
            "rank": 24.145872,
            "rank2": 5.944285869598389,
            "globalScore": 32.28855,
            "globalScoreReducedCityBoost": 30.0,
            "globalScoreWOHotelCountNormalize": 32.097088,
            "country": "US",
            "address": null,
            "zip": null,
            "countryName": "UNITED STATES",
            "countryCode": "US",
            "poiCategoryTypeId": 10,
            "poiCategoryName": "Famous Streets",
            "poiID": null,
            "seType": null,
            "gmtOffset": -8.0,
            "entered": "los angeles downtown",
            "highlightedName": null,
            "displayLine1": "Downtown Los Angeles",
            "displayLine2": "Los Angeles, California, United States",
            "fromSavedSearch": false
        },
...
```
Lat, long are not actually used by the search form but might just be part of the location object sent to the backend after the search button is clicked.

Reply:
Regarding the dropdown list, it could be a different data/state. The main purpose of the form data is to collect and submit user inputs, even if the input field will have a list of selections returned by backend, at the end once the user select one from the list, it will still show the format of "city, state". The extra fields don't need to be saved in the form data. 

Regarding the request sent to the backend search, user input can be used to filter the backend response, either based on city, state or neigh hood, but as long as we get a match, we can always use its id that is corresponding to the backend data object. I don't think lat/long needs to be sent back. I didn't spend time in digging into the network activities, but by briefly looking at the query URL, I think this is how Priceline behaves now. The hotel list page url: https://www.priceline.com/relax/in/3000009211/from/20241125/to/20241126/rooms/1/adults/2 ->
in/<regin id>/from/<checkin date>/to/<checkout date>/rooms/<rooms count>/adults/<adults count>
<img width="1497" alt="image" src="https://github.com/user-attachments/assets/6ac7e17b-ea10-4bb2-95a9-323698c01681">